pkg/client/record/event.go

```diff
@@ -145,7 +145,11 @@ func recordEvent(sink EventSink, event *api.Event, updateExistingEvent bool) boo
 	var err error
 	if updateExistingEvent {
 		newEvent, err = sink.Update(event)
-	} else {
+	}
+	// Update can fail because the event may have been removed and it no longer exists.
+	if !updateExistingEvent || (updateExistingEvent && err != nil) {
+		// Making sure that ResourceVersion is empty on creation
```

Comment:
Best we can do is to check if err code is 500 because apiserver is not returning a more specific error than that.